### ISSUE TITLE
Unit тесты должны тестировать только один модуль (у тебя почти всегда модулем является метод)

### ISSUE THREAD

**Issue by Znack** [ktmn17/game-of-life#12]
Например, в тестах `lengthHandler` ты сначала эмитишь событие, а потом смотришь, что в DOM изменились какие-то показатели. Но DOM у тебя меняет вью, который запускает перерисовку из метода `drawInitialCells` который в свою очередь только вызывается в обработчике onblur. Так вот - проще через синон шпиона повесить на `drawInitialCells` и смотреть, что метод был вызван, а не проверять разметку напрямую (подумай, что например у тебя тест необоснованно сломается, когда `drawInitialCells` вдруг будет некорректный, хотя `lengthHandler` сделал все как надо)

**Comment by ktmn17** [comment]
Там так и сделано. Проверка идет отдельно, что нарисовалось не больше 40 линий. Так что даже, если здесь будет ошибка, остальное нормально протестирует.
 ```
this.length.value <= 40 ? true : this.length.value = 40;
``` 
Эту строку конечно надо во вью перенести и там тестировать.

Остальное в контроллере проверяет именно вызовы методов.

**Comment by Znack** [comment]
Если честно, я так и не понял, понял ли ты меня)
Я говорил про то, что в тестах метода lengthHandler надо замокать все функции, которые там вызываются, чтобы на их надежность не полагаться и от их реализации не зависеть

**Comment by ktmn17** [comment]
То есть не только проверять вызовы внутренних функций (что у меня реализовано), но также ставить заглушки на эти функции, чтобы их вызов был фиктивным

**Comment by Znack** [comment]
Ну да, сделай как ты это предполагаешь, я посмотрю.

**Comment by ktmn17** [comment]
Я так понимаю, если я тестирую метод, то все внутренние функции, необходимые для выполнения метода, я должен псевдовызывать, с возвратом фиксированного значения, которая эта функция должна дать для проверяемого случая, записываемое через sinon.stub. Ладно, когда функция возвращает тру, фолс, числа, строки - это можно сымитировать, а если функция возвращает массив с 1000 случайных показателей, либо большой объект? Это явно не вручную прописывают и без вызова внутренней функции не обойтись.

**Comment by ktmn17** [comment]
Обновил тесты модели, все методы тестируются не зависимо друг от друга. Единственное, если я наследую класс внутри метода, непонятно как его мокать

**Comment by Znack** [comment]
На самом деле если ты приложение делаешь модульное, и тестируешь только отдельные методы, то там несложно замокать будет и сложные конструкции. Тем более это все-таки довольно редко надо будет делать. В общем, на практике это не так страшно, и однозначно оно того стоит

**Comment by Znack** [comment]
А зачем ты удалил тесты контроллера и вью?

**Comment by ktmn17** [comment]
Их подкорректировать надо. Не понятно как обьявления класса здесь мокать, видимо надо код менять, чтобы это замокать можно было

**Comment by Znack** [comment]
http://stackoverflow.com/a/38414108/1573086
Вот здесь глянь, как замокать класс Cell

**Comment by ktmn17** [comment]
https://github.com/ktmn17/game-of-life/commit/76e211b036ea5edbb9deb75dfea1024268de44e6 Теперь все тесты в контроллере не зависимые.

**Comment by Znack** [comment]
Так ты шпионов на setAlive и setDead так и не повешал?

**Comment by ktmn17** [comment]
Нет, для чего? Проверяю конечный результат. Импорт Cell заменяется внутренним классом TestCell

**Comment by ktmn17** [comment]
Обновил все тесты https://github.com/ktmn17/game-of-life/commit/881a85efa7758c1c2b1d1a9f16a80213816397eb
